src/Database/Kioku/Internal/TrieIndex.hs

```diff
@@ -23,22 +24,37 @@ import            System.IO
 import            Database.Kioku.Internal.Buffer
 import            Database.Kioku.Memorizable
 
-
 trieLookup :: BS.ByteString -> TrieIndex -> [Int]
 trieLookup key = maybe [] trieRootElems . lookupSubtrie key
 
 trieMatch :: BS.ByteString -> TrieIndex -> [Int]
 trieMatch prefix = maybe [] trieElems . lookupSubtrie prefix
 
+trieAllHitsAlong :: BS.ByteString -> TrieIndex -> [Int]
+trieAllHitsAlong "" _ = []
+trieAllHitsAlong key trie =
+  let (p,ath) = BS.splitAt 1 key
+  in concat $ go p ath trie
```

Comment:
I'm always suspicious about `concat` performance. Is there a clever way to avoid it here?